c++/src/capnp/rpc-twoparty-test.c++

```diff
@@ -738,6 +738,118 @@ KJ_TEST("Streaming over RPC") {
   }
 }
 
+KJ_TEST("Streaming over a chain of local and remote RPC calls") {
+  // This test verifies that a local RPC call that eventually resolves to a remote RPC call will
+  // still support streaming calls over the remote connection.
+
+  class TestStreamCalleeImpl final: public test::TestStreamCallee::Server {
+  public:
+    kj::Maybe<TestStreamingImpl&> streamingImpl;
+
+    kj::Promise<void> getStreamingCap(GetStreamingCapContext context) override {
+      context.releaseParams();
+
+      auto result = context.getResults();
+      auto s = kj::heap<TestStreamingImpl>();
+      KJ_ASSERT(streamingImpl == nullptr);
+      streamingImpl = *s;
+      result.setCap(kj::mv(s));
+
+      return kj::READY_NOW;
+    }
+  };
+
+  class TestStreamCallerImpl final: public test::TestStreamCaller::Server {
+  public:
+    explicit TestStreamCallerImpl(
+        kj::Promise<test::TestStreamCallee::Client> calleePromise)
+        : calleePromise(kj::mv(calleePromise)) {}
+
+    kj::Maybe<kj::Promise<test::TestStreamCallee::Client>> calleePromise;
+
+    kj::Promise<void> getCallee(GetCalleeContext context) override {
+      context.releaseParams();
+
+      auto result = context.getResults();
+      result.setCap(kj::mv(KJ_ASSERT_NONNULL(calleePromise)));
+      calleePromise = nullptr;
+
+      return kj::READY_NOW;
+    }
+  };
+
+  kj::EventLoop loop;
+  kj::WaitScope waitScope(loop);
+
+  // Set up a local "caller" server that eventually delegates requests for callees to a remote
```

Comment:
FWIW, I think this test could be simplified by using a PromiseFulfillerPair here, like:

```
auto paf = kj::newPromiseAndFulfiller<test::TestStreaming::Client>();
test::TestStreaming::Client promisedClient = kj::mv(paf.promise);
```

Then:

1. make a streaming call to promisedClient, verify it doesn't resolve because we're blocked on a local promise
2. resolve the paf to a remote capability, but where the remote cap's method implementation hangs or otherwise doesn't return immediately
3. now verify that the original streaming call's returned promise has resolved, and several more streaming calls can be made, until the window fills up

That said, if you'd rather just leave the test as-is and move on I'm also fine with that.

Reply:
Thanks, that seems much simpler.